rest/models/pom.xml

```diff
@@ -27,33 +27,20 @@
             <version>${project.version}</version>
             <type>zip</type>
         </dependency>
-        <dependency>
-            <groupId>org.immutables</groupId>
-            <artifactId>value</artifactId>
-        </dependency>
-        <dependency>
-            <groupId>org.slf4j</groupId>
-            <artifactId>slf4j-api</artifactId>
-        </dependency>
-        <dependency>
-            <groupId>com.google.guava</groupId>
-            <artifactId>guava</artifactId>
-        </dependency>
         <dependency>
             <groupId>javax.annotation</groupId>
             <artifactId>javax.annotation-api</artifactId>
+            <version>${annotation.api.version}</version>
```

Comment:
❓ Why are we opting for defining versions in child poms? I know we had to do that with Karaf and OSGi for packaging reasons but this approach goes against maven's design where dependency management in the parent pom should be where versions and exclusions of the sort be defined and child poms simply specify the dependency and inherit those. In general, versions should never be defined in child poms unless there is a strict reason why a different version is required.

Reply:
Do you recommend using properties to define versions in the root POM or using Maven dependency management to define versions?

Reply:
Yes. Using properties but only referencing them from within the dependency management section. There is no need to replicate the version info throughout all pom files as you would not normally replicate the scope unless there is a need to change it. This is the whole reason for the dependency management section. I realized that with Karaf and how features were packaged we were forced to follow the paradigm of hardcoding the version (via a property as it is better than actually hardcoding), but this is no longer required now and we should move away from that practice that goes against how Maven is designed to work.